contracts/upgradeable_contracts/arbitrary_message/BasicAMB.sol

```diff
@@ -0,0 +1,30 @@
+pragma solidity 0.4.24;
+
+import "../../upgradeability/EternalStorage.sol";
+import "../../libraries/SafeMath.sol";
+import "../Validatable.sol";
+import "../BasicBridge.sol";
+
+
+contract BasicAMB is BasicBridge {
+    bytes constant internal SUBSIDIZED_MODE = bytes(abi.encodePacked("AMB-subsidized-mode"));
+    bytes constant internal DEFRAYAL_MODE = bytes(abi.encodePacked("AMB-defrayal-mode"));
+
+    function setSubsidizedModeForForeign() public onlyOwner {
+        bytesStorage[keccak256(abi.encodePacked("foreignBridgeMode"))] = SUBSIDIZED_MODE;
```

Comment:
Since it is `BasicAMB` which means to be generic, it should not contain a setter and a getter for `foreignBridgeMode`

Reply:
The getters and setters are for the bridge mode which will be needed on both contracts to know how to pass the messages. I updated the name from `foreignBridgeMode` to `homeToForeignMode` so it's easier to understand.